src/common/os_linux.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2017, Intel Corporation
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in
+ *       the documentation and/or other materials provided with the
+ *       distribution.
+ *
+ *     * Neither the name of the copyright holder nor the names of its
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * os_linux.h -- Linux specific API
+ */
+
+#include <stdint.h>
+
+struct os_extent_iter;
+
+struct os_extent {
+	uint64_t offset;
+	uint64_t length;
+};
+
+struct os_extent_iter *os_extent_new(int fd);
+void os_extent_delete(struct os_extent_iter *iter);
+
+int os_extent_next(struct os_extent_iter *iter, struct os_extent *extent);
+
+struct os_sysfs_iter;
+
+struct os_sysfs_iter *os_sysfs_new(const char *path, const char *format);
+struct os_sysfs_iter *os_sysfs_dev_new(int fd, const char *subpath,
+	const char *format);
+
+void os_sysfs_delete(struct os_sysfs_iter *iter);
+int os_sysfs_next(struct os_sysfs_iter *iter, ...);
+
+int os_sysfs_single(const char *path, const char *format, ...);
+int os_sysfs_dev_single(int fd, const char *subpath, const char *format, ...);
```

Comment:
I'm open to suggestions.
So, instead of a single file, we would have:
extent.h
extent_linux.c
sysfs.h
sysfs.c

?